### PR TITLE
[Qt5] Update to 5.12.7 and add missing qt submodules

### PR DESCRIPTION
**Pull request by Neumann-A**
merge #9705 first: 

~~currently: mainly CI run to get the missing hashes~~

probably will break latest feature which will be fixed by an upcoming PR updating latest to 5.14.1.

### CODE REVIEW

File: ports/qt5-androidextras/CONTROL [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Source: qt5-androidextras
+Version: 5.12.5-2
```
Comment (Neumann-A) [inline ports/qt5-androidextras/CONTROL]: needs update.

File: ports/qt5-base/portfile.cmake [hunk 2]
```diff
@@ -299,12 +303,28 @@ else()
...
+    )]] 
+[[    set_target_properties(Qt5::WinMain PROPERTIES
+        IMPORTED_LOCATION_RELEASE ${imported_location_release}
+        IMPORTED_LOCATION_DEBUG ${imported_location_debug}
+    )]] 
+    _contents "${_contents}")
+        file(WRITE "${cmakefile}" "${_contents}")
+    endif()
+    
     file(COPY ${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake DESTINATION ${CURRENT_PACKAGES_DIR}/share/qt5core)
```
Comment (Neumann-A) [inline ports/qt5-base/portfile.cmake]: remove if CI run is succesful

File: ports/qt5-canvas3d/CONTROL [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Source: qt5-canvas3d
+Version: 5.12.5-2
```
Comment (Neumann-A) [inline ports/qt5-canvas3d/CONTROL]: update

File: ports/qt5-doc/CONTROL [hunk 4]
```diff
@@ -0,0 +1,4 @@
+Source: qt5-doc
+Version: 5.12.5-2
```
Comment (Neumann-A) [inline ports/qt5-doc/CONTROL]: update

File: ports/qt5-serialbus/CONTROL [hunk 5]
```diff
@@ -0,0 +1,4 @@
+Source: qt5-serialbus
+Version: 5.12.5-2
```
Comment (Neumann-A) [inline ports/qt5-serialbus/CONTROL]: update

File: ports/qt5-translations/CONTROL [hunk 6]
```diff
@@ -0,0 +1,4 @@
+Source: qt5-translations
+Version: 5.12.5-2
```
Comment (Neumann-A) [inline ports/qt5-translations/CONTROL]: update

File: ports/qt5-wayland/CONTROL [hunk 7]
```diff
@@ -0,0 +1,4 @@
+Source: qt5-wayland
+Version: 5.12.5-2
```
Comment (Neumann-A) [inline ports/qt5-wayland/CONTROL]: update

File: ports/qt5-webengine/CONTROL [hunk 8]
```diff
@@ -0,0 +1,4 @@
+Source: qt5-webengine
+Version: 5.12.5-2
```
Comment (Neumann-A) [inline ports/qt5-webengine/CONTROL]: update

File: ports/qt5-webglplugin/CONTROL [hunk 9]
```diff
@@ -0,0 +1,4 @@
+Source: qt5-webglplugin
+Version: 5.12.5-2
```
Comment (Neumann-A) [inline ports/qt5-webglplugin/CONTROL]: update

File: ports/qt5-x11extras/CONTROL [hunk 10]
```diff
@@ -0,0 +1,4 @@
+Source: qt5-x11extras
+Version: 5.12.5-2
```
Comment (Neumann-A) [inline ports/qt5-x11extras/CONTROL]: update

File: ports/qt5/CONTROL [hunk 11]
```diff
@@ -2,8 +2,173 @@ Source: qt5
 Version: 5.12.5
```
Comment (Neumann-A) [inline ports/qt5/CONTROL]: update

**Comment by Neumann-A** [comment]
```
ERROR: Feature 'xcb' was enabled, but the pre-condition 'features.thread && features.xkbcommon && libs.xcb' failed.

ERROR: Feature 'system-xcb' was enabled, but the pre-condition 'features.xcb && tests.xcb_syslibs' failed.
```

Reason: `features.xkbcommon` can not be found. This blocks `qt5-x11extras`. Please install `xkbcommon`

**Comment by Neumann-A** [comment]
hmm qtkeychan:
```
CMake Error at C:/vsts/_work/3/s/installed/x64-windows-static/share/cmake/Qt5Core/Qt5CoreMacros.cmake:131 (file):
  file failed to open for writing (Invalid argument):

    C:/vsts/_work/3/s/buildtrees/qtkeychain/x64-windows-static-rel/moc_C:/vsts/_work/3/s/buildtrees/qtkeychain/x64-windows-static-rel/keychain.cpp_parameters
Call Stack (most recent call first):
  C:/vsts/_work/3/s/installed/x64-windows-static/share/cmake/Qt5Core/Qt5CoreMacros.cmake:181 (qt5_create_moc_command)
  CMakeLists.txt:65 (qt5_wrap_cpp)
  CMakeLists.txt:168 (QT_WRAP_CPP)
```

Did they change the moc behavior because this is most certainly wrong:
`    C:/vsts/_work/3/s/buildtrees/qtkeychain/x64-windows-static-rel/moc_C:/vsts/_work/3/s/buildtrees/qtkeychain/x64-windows-static-rel/keychain.cpp_parameters`

File: ports/qt5-base/patches/prl_parser.patch [hunk 12]
```diff
@@ -0,0 +1,260 @@
...
++          list(APPEND _lib_deps ${_flag}) # changed from _link_flags
++      else()
++          # Handle all remaining flags by simply passing them to the linker
++          #message(STATUS \"Added to link flags (${Configuration}): ${_flag}\")
++          list(APPEND _lib_flags ${_flag}) # changed from _link_flags
++      endif()
++    endforeach()
++    
++    message(STATUS \"${_qt_prl_TARGET} links (via PRL ${_qt_prl_CONFIGURATION}): ${_lib_deps}\")
++    message(STATUS \"${_qt_prl_TARGET} has linker flags (via PRL): ${_lib_flags}\")
```
Comment (Neumann-A) [inline ports/qt5-base/patches/prl_parser.patch]: need to comment all the debug linkage messages out before final merge

**Comment by Neumann-A** [comment]
Anybody want to make qt5-webengine work? Somehow the build fails due to not finding an include which if checked exists. Don't know whats happening here? Maybe the paths are getting too long?

**Comment by pwilkin** [comment]
@Neumann-A There is a but in qt-webengine with precompiled headers (https://bugreports.qt.io/browse/QTBUG-65677). I've tried to include the workarounds they mention there, however, I can't get it to compile, it still seems to be including the precompiled headers for whatever reason.

**Comment by pwilkin** [comment]
For me, this doesn't work - when building 5.12.7, the common.pri file lands not in src/buildtools, but in src/core and the line numbers are different, I had to modify the patch to:

```
diff --git a/src/core/config/common.pri b/src/core/config/common.pri
index 97d39535c..dc09da43a 100644
--- a/src/core/config/common.pri	
+++ b/src/core/config/common.pri
@@ -55,6 +55,6 @@
 
 precompile_header {
-    gn_args += enable_precompiled_headers=true
+    gn_args += enable_precompiled_headers=false
 } else {
     gn_args += enable_precompiled_headers=false
 }
```

**Comment by pwilkin** [comment]
Disregard, just saw the new commit :>

**Comment by Neumann-A** [comment]
Build will still fail with it.... at somewhere arround 22300 of 22900 but with a different error

**Comment by pwilkin** [comment]
Is there any way to enable incremental compilation after failure? This package is so bloated it's not even funny, on my i7-9700K with 24GB RAM and an SSD drive the build takes about half an hour...

**Comment by Neumann-A** [comment]
You could manually open a VS dev prompt and run ninja/nmake in the builddir. I sometimes do that to try out quick fixes

**Comment by pwilkin** [comment]
I saw your fix for the new issue on the QT mailing list, it seems to work (at least the debug build has succeeded, building rel now).

**Comment by pwilkin** [comment]
Ran out of disk space, apparently 117 GB (!) is not enough for a full compilation. Is there any way to optimize this build process?

**Comment by Neumann-A** [comment]
CI also ran out of disk space ;) so you are not alone.
But the question how to reduce that should probably be asked elsewhere (like the qt dev team?)

**Comment by Neumann-A** [comment]
closing in favor for #10644

**Pull request closed without merging**